### PR TITLE
add ledvance fw provider

### PR DESCRIPTION
**Pull request by billyburly**
Adds provider to support Ledvance zigbee firmware updates. Provides updates for Sylvania, Osram, etc. Tested against Osram Lightify Gardenspot RGB.

### CODE REVIEW

**Comment by Adminiuga** [comment]
Thanks! I've glanced over and looks good. Will take a better look when have more time.

File: zigpy/ota/provider.py [hunk 1]
```diff
@@ -161,6 +162,101 @@ class Trådfri(Basic):
...
+        async with aiohttp.ClientSession() as req:
+            LOGGER.debug("Downloading %s for %s", self.url, self.key)
+            async with req.get(self.url) as rsp:
+                data = await rsp.read()
+
+        assert len(data) > 24
+        offset = data.index(self.OTA_HEADER)
+        assert offset >= 0
+
+        img, _ = OTAImage.deserialize(data[offset:])
```
Comment (Adminiuga) [inline zigpy/ota/provider.py]: Are Ledvance OTA images not starting with the OTA header? the Ikea images were not standard, wrapped in another container and needed the checks above. If they are standard OTA images you can skip lines 203--205 and just do `img, _ = OTAImage.deserialize(data)`

Comment (billyburly) [reply zigpy/ota/provider.py]: The Ledvance images do start with the header. Like you inferred, I followed the IKEA provider. Will remove the check

Comment (Adminiuga) [reply zigpy/ota/provider.py]: Don't worry about it for now.

**Pull request merged**